### PR TITLE
CAMEL-17774 Added check for null inputStream

### PR DESCRIPTION
**Pull request by tsantakismanos**
This PR is to add a check for null during `ZipIterator` creation which will prevent the assignment of a `new ZipInputStream(new BufferedInputStream(inputStream))` to `zipInputStream` variable. 
Such case may occur when a message is converted to `InputStream` but no `TypeConverter` is present (e.g in `ZipSplitter`)

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipIterator.java [hunk 1]
```diff
@@ -48,6 +48,11 @@
     public ZipIterator(Exchange exchange, InputStream inputStream) {
         this.exchange = exchange;
         this.allowEmptyDirectory = false;
+
+        if (inputStream == null) {
+            throw new IllegalStateException("Input stream cannot be null");
+        }
```
Comment (essobedo) [inline components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipIterator.java:54]: I’m not sure that an IllegalStateException is the right choice in this case, I would rather use ´Objects.requireNonNull´ Instead.

Comment (tsantakismanos) [reply components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipIterator.java]: Indeed , changed to the proposed one.

**Comment by oscerd** [comment]
In case you'll need to check a null value, you could have a look at ObjectHelper class methods. It's not for this case, but it might help for future PRs.

**Comment by davsclaus** [comment]
Thanks you are welcome to look into camel-tarfile if this component also should have such a null check too.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**